api/dataprocessor_test.go

```diff
@@ -1095,6 +1096,194 @@ func TestMergeSeries(t *testing.T) {
+	req := reqRaw(metric, 44, 88, 100, 10, consolidation.None)
+	req.ArchInterval = archInterval
+	ctx := newRequestContext(&req, consolidation.None)
+
+	expectFrom := uint32(41)
+	if ctx.From != expectFrom {
+		t.Errorf("requestContext.From is not at boundary as expected, expected/got %d/%d", expectFrom, ctx.From)
+	}
+
+	expectTo := uint32(81)
+	if ctx.To != expectTo {
+		t.Errorf("requestContext.To is not at boundary as expected, expected/got %d/%d", expectTo, ctx.To)
+	}
+
+	expectedAggKey := ""
+	if ctx.AggKey != expectedAggKey {
+		t.Errorf("requestContext.AggKey is not empty as expected, expected/got \"%s\"/\"%s\"", expectedAggKey, ctx.AggKey)
+	}
+}
+
+func TestRequestContextWithConsolidator(t *testing.T) {
+	metric := "metric1"
+	archInterval := uint32(10)
+	from := uint32(44)
+	to := uint32(88)
+	req := reqRaw(metric, from, to, 100, 10, consolidation.Sum)
+	req.ArchInterval = archInterval
+	ctx := newRequestContext(&req, consolidation.Sum)
+
+	expectFrom := from
+	if ctx.From != expectFrom {
+		t.Errorf("requestContext.From is not original value as expected, expected/got %d/%d", expectFrom, ctx.From)
+	}
+
+	expectTo := to
+	if ctx.To != expectTo {
+		t.Errorf("requestContext.To is not original value as expected, expected/got %d/%d", expectTo, ctx.To)
+	}
+
+	expectedAggKey := fmt.Sprintf("%s_%s_%d", metric, "sum", archInterval)
+	if ctx.AggKey != expectedAggKey {
+		t.Errorf("requestContext.AggKey is not as expected, expected/got \"%s\"/\"%s\"", expectedAggKey, ctx.AggKey)
+	}
+}
+
+func TestGetSeriesCachedStore(t *testing.T) {
+	chunkSpan := uint32(600)
+	numChunks := uint32(10)
+	stats, _ := helper.New(false, "", "standard", "metrictank", "")
+	srv, _ := NewServer(stats)
+	store := mdata.NewMockStore()
+	srv.BindBackendStore(store)
+	metrics := mdata.NewAggMetrics(store, chunkSpan, numChunks, 0, 0, 0, 0, []mdata.AggSetting{})
+	mdata.InitMetrics(stats)
+	srv.BindMemoryStore(metrics)
+	defer metrics.Stop()
+	c := cache.NewCCache()
+	srv.BindCache(c)
+	metricKey := "metric1"
+
+	// generating 4 chunks
+	chunks := make([]*chunk.Chunk, 0)
+	for i := chunkSpan; i < chunkSpan*5; i++ {
+		if i%chunkSpan == 0 {
+			chunks = append(chunks, chunk.New(i))
+		}
+		chunks[len(chunks)-1].Push(i, float64(i*2))
+	}
+
+	prevts := uint32(0)
+	// adding 3 itgens to cache
+	var i int
+	for i = 0; i < 3; i++ {
+		chunks[i].Series.Finish()
+		itgen := chunk.NewBareIterGen(chunks[i].Series.Bytes(), chunks[i].Series.T0, chunkSpan)
+		srv.Cache.Add(metricKey, prevts, *itgen)
+		prevts = itgen.Ts()
+	}
+
+	// preparing the next itgen as store mock result (to simulate one that has just been added and not cached yet)
+	itgens := make([]chunk.IterGen, 0)
+	chunks[i].Series.Finish()
+	itgens = append(itgens, *chunk.NewBareIterGen(chunks[i].Series.Bytes(), chunks[i].Series.T0, chunkSpan))
+
+	steps := chunkSpan / 10
+	for from := uint32(chunkSpan + chunkSpan/2); from < uint32(chunkSpan*4); from += steps {
+		for to := from + 1; to <= from+chunkSpan+1; to += steps {
```

Comment:
can you explain the choice of range for from and to that we're testing here, and how it relates to the data in the chunks?
it looks like it's well chosen (to cover a variety of cases), but a comment would clarify